File: Data Structures & Algorithms/binary-search/submission-2.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int f = 0, e = nums.size();
        int mid = (f+e)/2;
        if (nums.empty()) {
            return -1;
        }
        if(nums[mid]==target){
            return mid;
        }
        if(nums[mid]<target){
            
            vector <int> sub(nums.begin()+mid+1, nums.end());
            int res = search(sub,target);
            if(res==-1){
                return -1;
            }
            return res+mid+1;
        }
        else{
            vector <int> sub(nums.begin(), nums.begin()+mid);
            return search(sub, target);
        }
        return -1;
    }
};

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-2.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        l, r = 0, 0
        res = float('inf')
        prefix = 0
        while r < len(nums):
            prefix += nums[r]
            while prefix >= target:
                res = min(res, r - l +1)
                prefix-=nums[l]
                l+=1
            r+=1
        return res if res != float('inf') else 0
